connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/processors/recipients/RecipientsFilter.java

```diff
@@ -0,0 +1,59 @@
+package com.redhat.cloud.notifications.connector.email.processors.recipients;
+
+import com.redhat.cloud.notifications.connector.email.constants.ExchangeProperty;
+import com.redhat.cloud.notifications.connector.email.model.User;
+import com.redhat.cloud.notifications.connector.email.model.settings.RecipientSettings;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class RecipientsFilter implements Processor {
+    /**
+     * Filters the fetched users according to the criteria specified in the
+     * recipient settings.
+     * @param exchange the exchange of the pipeline.
+     */
+    @Override
+    public void process(final Exchange exchange) {
+        // Fetch the required data to filter the users.
+        final RecipientSettings recipientSettings = exchange.getProperty(ExchangeProperty.CURRENT_RECIPIENT_SETTINGS, RecipientSettings.class);
+        final Set<String> subscribers = (Set<String>) exchange.getProperty(ExchangeProperty.SUBSCRIBERS, Set.class);
+        Set<User> users = (Set<User>) exchange.getProperty(ExchangeProperty.USERS, Set.class);
```

Comment:
I'll write a comment here but it applies to several places in the code.

I don't think we need to store a set of `User`s in the exchange because we're only using the `User#username` field while sending the emails. `User#isAdmin` and `User#isActive` may look to be needed but they're not. We can check is a user is active or is an admin without relying on the `User` class at all. In the end, the only field that matters is `User#username`, so we could replace `Set<User>` with `Set<String>` and put the usernames in that set.

Reply:
In other words, the `User` class could be deleted from the connector and everything would still work just fine.